xcm/pallet-xcm/src/lib.rs

```diff
@@ -553,16 +569,54 @@ pub mod pallet {
 		}
 	}
 
+	impl<T: Config> DropAssets for Pallet<T> {
+		fn drop_assets(origin: &MultiLocation, assets: Assets) -> Weight {
+			if assets.is_empty() {
+				return 0
+			}
+			let versioned = VersionedMultiAssets::from(MultiAssets::from(assets));
+			let hash = BlakeTwo256::hash_of(&(&origin, &versioned));
+			AssetTraps::<T>::mutate(hash, |n| *n += 1);
+			Self::deposit_event(Event::AssetsTrapped(hash, origin.clone(), versioned));
```

Comment:
This event becomes really, really important if people want to look for their own assets that were trapped inside of the pallet. Wonder if it's a good idea to allow some amount of querying by the sender into the map to see how much assets were trapped?

Reply:
The map only stores hashes, by design. If they care, they'll need to check the event.